Rocket.Chat/Controllers/Chat/MessagesListViewController.swift

```diff
@@ -54,41 +65,62 @@ class MessagesListViewData {
         if let subscription = subscription {
             isLoadingMoreMessages = true
 
-            let request = SubscriptionMessagesRequest(roomId: subscription.rid, type: subscription.type, query: query)
             let options = APIRequestOptions.paginated(count: pageSize, offset: currentPage*pageSize)
+            if isListingMentions {
+                let request = SubscriptionMentionsRequest(roomId: subscription.rid)
+                API.current()?.fetch(request, options: options, succeeded: { result in
+                    self.handleMessages(
+                        fetchingWith: result.fetchMessagesFromRealm,
+                        showing: result.count,
+                        total: result.total,
+                        completion: completion
+                    )
+                }, errored: { _ in
+                    Alert.defaultError.present()
+                })
+            } else {
+                let request = SubscriptionMessagesRequest(roomId: subscription.rid, type: subscription.type, query: query)
+                API.current()?.fetch(request, options: options, succeeded: { result in
```

Comment:
Looks like the call of the `fetch()` could be the same for both requests?

Reply:
@rafaelks Since we have two different requests but they're both `APIRequest` you probably are thinking that we could just initialize a generic `var request: APIRequest` with the generic type based on the condition, but we can't use it to call `API.fetch` because it would not be able to identify which generic type we would be calling and it would throw a compilation error saying that the `API.fetch` call is ambiguos.

Reply:
👍